### PR TITLE
Migrate to Java 21

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/producer_client_tests.bal [hunk 1]
```diff
@@ -180,7 +180,7 @@ function producerGetTopicPartitionsErrorTest() returns error? {
     check producer->close();
 }
 
-@test:Config {enable: true}
+@test:Config {enable: false}
```
Comment (warunalakshitha) [inline ballerina/tests/producer_client_tests.bal:183]: We can create a issue for this

Comment (HindujaB) [reply ballerina/tests/producer_client_tests.bal:183]: created https://github.com/ballerina-platform/ballerina-library/issues/7372

File: native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java [hunk 2]
```diff
@@ -81,4 +83,12 @@ public synchronized String format(LogRecord lr) {
         LogManager.getLogManager().addLogger(apacheKafkaLogger);
         LogManager.getLogManager().addLogger(balKafkaLogger);
     }
+
+    public static Object getResult(CompletableFuture<Object> balFuture) {
+        try {
+            return balFuture.get();
```
Comment (warunalakshitha) [inline native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java:89]: Need to fix interrupt issue

File: native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java [hunk 3]
```diff
@@ -81,4 +83,12 @@ public synchronized String format(LogRecord lr) {
...
+        } catch (Throwable throwable) {
+            throw ErrorCreator.createError(throwable);
+        }
+    }
```
Comment (warunalakshitha) [inline native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java:93]:
```suggestion
    public static Object getResult(CompletableFuture<Object> balFuture) {
        try {
            return balFuture.get();
        } catch (BError error) {
            throw error;
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            throw ErrorCreator.createError(e);
        } catch (Throwable throwable) {
            throw ErrorCreator.createError(throwable);
        }
    }
```

File: native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java [hunk 4]
```diff
@@ -20,8 +20,10 @@
 
 import io.ballerina.runtime.api.Environment;
 import io.ballerina.runtime.api.Module;
+import io.ballerina.runtime.api.creators.ErrorCreator;
```
Comment (warunalakshitha) [inline native/src/main/java/io/ballerina/stdlib/kafka/utils/ModuleUtils.java:23]:
```suggestion
import io.ballerina.runtime.api.creators.ErrorCreator;
import io.ballerina.runtime.api.values.BError;
```

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**